core/src/main/java/org/apache/iceberg/metrics/ScanReportParser.java

```diff
@@ -107,14 +117,20 @@ public static ScanReport fromJson(JsonNode json) {
     List<String> projectedFieldNames = JsonUtil.getStringList(PROJECTED_FIELD_NAMES, json);
     ScanMetricsResult scanMetricsResult =
         ScanMetricsResultParser.fromJson(JsonUtil.get(METRICS, json));
-    return ImmutableScanReport.builder()
-        .tableName(tableName)
-        .snapshotId(snapshotId)
-        .schemaId(schemaId)
-        .projectedFieldIds(projectedFieldIds)
-        .projectedFieldNames(projectedFieldNames)
-        .filter(filter)
-        .scanMetrics(scanMetricsResult)
-        .build();
+    ImmutableScanReport.Builder builder =
+        ImmutableScanReport.builder()
+            .tableName(tableName)
+            .snapshotId(snapshotId)
+            .schemaId(schemaId)
+            .projectedFieldIds(projectedFieldIds)
+            .projectedFieldNames(projectedFieldNames)
+            .filter(filter)
+            .scanMetrics(scanMetricsResult);
+
+    if (json.has(METADATA)) {
+      builder.metadata(JsonUtil.getStringMap(METADATA, json));
```

Comment:
Isn't there a `getStringMapOrNull` method?

Reply:
there isn't, but even if there would be a `getStringMapOrNull()`, we don't want to set `metadata(null)` as that will fail